pkg/discord/discord.go

```diff
@@ -0,0 +1,113 @@
+package discord
+
+import (
+	"fmt"
+	"os"
+	"strings"
+	"time"
+
+	"github.com/bwmarrin/discordgo"
+	"github.com/caarlos0/env/v9"
+	"github.com/rs/zerolog"
+
+	"github.com/warden-protocol/discord-faucet/pkg/faucet"
+)
+
+type Discord struct {
+	Session  *discordgo.Session
+	MC       *discordgo.MessageCreate
+	Token    string `env:"TOKEN" envDefault:""`
+	Requests map[string]time.Time
+	Faucet   faucet.Faucet
+	logger   zerolog.Logger
+}
+
+func InitDiscord() (Discord, error) {
+	var err error
+
+	d := Discord{}
+	if err = env.Parse(&d); err != nil {
+		return Discord{}, err
+	}
+
+	d.logger = zerolog.New(
+		zerolog.ConsoleWriter{Out: os.Stderr, TimeFormat: time.RFC3339},
+	).Level(zerolog.InfoLevel).With().Timestamp().Caller().Logger()
+
+	if d.Token == "" {
+		return Discord{}, fmt.Errorf("missing discord token")
+	}
+	d.Session, err = discordgo.New("Bot " + d.Token)
+	if err != nil {
+		return Discord{}, err
+	}
+
+	d.Requests = make(map[string]time.Time)
+
+	d.Faucet, err = faucet.InitFaucet()
+	if err != nil {
+		return Discord{}, err
+	}
+	d.Faucet.Logger = d.logger
+
+	return d, nil
+}
+
+func (d *Discord) MessageCreate(s *discordgo.Session, m *discordgo.MessageCreate) {
+	// Ignore all messages created by the bot itself
+	// This isn't required in this specific example but it's a good practice.
+	if m.Author.ID == s.State.User.ID {
+		return
+	}
+
+	d.Session = s
+	d.MC = m
```

Comment:
what are those for?

Reply:
You'r right, from some documentation this was needed but it can work without it.